dfsclient/src/test/java/org/hpccsystems/dfs/client/DFSRecordBuilderTest.java

```diff
@@ -0,0 +1,62 @@
+package org.hpccsystems.dfs.client;
+
+import java.util.ArrayList;
+import java.math.BigDecimal;
+
+import org.hpccsystems.dfs.client.IRecordBuilder;
+import org.hpccsystems.dfs.client.HPCCRecord;
+import org.hpccsystems.dfs.client.HPCCRecordBuilder;
+import org.hpccsystems.dfs.client.DataPartition;
+
+import org.hpccsystems.commons.ecl.FieldType;
+import org.hpccsystems.commons.ecl.FieldDef;
+import org.hpccsystems.commons.ecl.DummyFieldDefinitions;
+
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+public class DFSRecordBuilderTest
+{
+    private FieldDef testFieldDefinition = null;
+    private IRecordReader testRecordReader = null;
+
+    @Before
+    public void setup()
+    {
+        try {
+            testFieldDefinition = DummyFieldDefinitions.getComplexRecordDefinition();
+            testRecordReader = new DummyRecordReader(10);
+        } catch(Exception e) {
+            Assert.fail("Test setup failed with error: " + e.getMessage());
+        }
+    }
+
+    @Test
+    public void hpccRecordBuilderTest() throws Exception
+    {
+        try {
+            HPCCRecordBuilder recordBuilder = new HPCCRecordBuilder(testFieldDefinition);
+            testRecordReader.initialize(recordBuilder);
+
+            while (testRecordReader.hasNext()) {
```

Comment:
are there any other HPCCRecbuilder methods we should test?

Reply:
the TestRecordReader will call all of the HpccRecordBuilder functions